internal/fs/inode/dir_test.go

```diff
@@ -491,6 +492,44 @@ func (t *DirTest) LookUpChild_TypeCaching() {
 	ExpectEq(dirObjName, result.Object.Name)
 }
 
+func (t *DirTest) LookUpChild_NonExistentTypeCache() {
+	// Enable enableNonexistentType for type cache
+	t.resetInode(false, true)
```

Comment:
Can we also add the test-cases when both implicit-directory and Nonexistent type are true?

Reply:
Added.